core/ipc/common/kafka/src/main/java/org/opennms/core/ipc/common/kafka/KafkaHealthCheck.java

```diff
@@ -33,44 +33,35 @@
 
 import org.apache.kafka.clients.admin.AdminClient;
 import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.opennms.core.health.api.Context;
 import org.opennms.core.health.api.HealthCheck;
 import org.opennms.core.health.api.Response;
 import org.opennms.core.health.api.Status;
 
 
 public class KafkaHealthCheck implements HealthCheck {
 
-    private OsgiKafkaConfigProvider osgiKafkaConfigProvider;
-    private OnmsKafkaConfigProvider kafkaConfigProvider;
-    // Differentiate Sink/RPC
+    private KafkaConfigProvider kafkaConfigProvider;
+    // Differentiate between Sink/RPC
     private final String type;
     private Properties kafkaConfig = new Properties();
 
-    public KafkaHealthCheck(OsgiKafkaConfigProvider kafkaConfigProvider, String type) {
-        this.osgiKafkaConfigProvider = kafkaConfigProvider;
+    public KafkaHealthCheck(KafkaConfigProvider kafkaConfigProvider, String type) {
+        this.kafkaConfigProvider = kafkaConfigProvider;
         this.type = type;
     }
 
-    public KafkaHealthCheck(String sysPropPrefix, String type) {
-        this.kafkaConfigProvider = new OnmsKafkaConfigProvider(sysPropPrefix);
-        this.type = type;
-    }
 
     @Override
     public String getDescription() {
         return "Connecting to Kafka from " + type ;
     }
 
     @Override
-    public Response perform() throws Exception {
-        if (osgiKafkaConfigProvider != null) {
-            kafkaConfig = osgiKafkaConfigProvider.getProperties();
-        }
-        if (kafkaConfigProvider != null) {
-            kafkaConfig = kafkaConfigProvider.getProperties();
-        }
-        // Default 1 sec should be sufficient, need changes to health check API to make it configurable.
-        kafkaConfig.put("request.timeout.ms", 1000);
+    public Response perform(Context context) throws Exception {
+        kafkaConfig = kafkaConfigProvider.getProperties();
```

Comment:
The change here, will it create a timeout response? To me it looks like it would create a failure response, which is probably not what is intended. In general a HealthCheck implementation should not worry about time outs, but an implementation of the HealthCheckService should, see https://github.com/OpenNMS/opennms/blob/7840afa2e7802c7520c9347fc35552595a767d7d/core/health/impl/src/main/java/org/opennms/core/health/impl/DefaultHealthCheckService.java#L127.

If a HealthCheck must implement a timeout itself, we should verify if the DefaultHealthCheckService is not enforcing the timeout correctly, before we do that.

Reply:
It will not create timeout response but here` listTopicsResult.names().get()`  itself is a future, so this timeout is for this future to exit.

Reply:
Okay, makes sense, but can we also encounter for timeout responses then?

Reply:
most likely command will timeout before the internal future timesout. So user will get timeout response.